### PR TITLE
feat(client): add mTLS config

### LINKED ISSUE #189: feat: mTLS configuration for HTTPS and TCP monitors
Please add the ability to specify client certificate & key, and server CA certificate, for monitoring of HTTPS & TLS endpoints that require mTLS handshakes (and/or use self-signed server certificates). For example: monitoring [Kafka Broker](https://docs.confluent.io/platform/current/kafka/configure-mds/mutual-tls-auth-rbac.html) listener endpoints. 

A couple similar examples:
 - https://github.com/louislam/uptime-kuma/issues/139
 - https://github.com/statping/statping/issues/597

### PR DESCRIPTION
**Pull request by chrisbsmith**
## Summary

closes #189 

Adds mTLS configuration to the client

## Checklist

- [x] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [x] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

**Review by appleboy — APPROVED** [review]

**Comment by chrisbsmith** [comment]
@TwiN just dropping a ping here to see if you have time to review. Thanks!

File: README.md
Comment (TwiN) [inline README.md:359]: `client` is redundant in the name since the parent configuration name is already `client`.

Also, I'd like this to be under `tls` so that it matches the existing `web.tls.certificate-file` and `web.tls.private-key-file` configuration, so in other words, the configuration would look like `client.tls.certificate-file` and `client.tls.private-key-file`.

File: README.md
Comment (TwiN) [inline README.md:409]:
```suggestion
Note that if running in a container, you must volume mount the certificate and key into the container.
```

**Review by TwiN — CHANGES_REQUESTED** [review]

File: README.md
Comment (chrisbsmith) [reply README.md]: Updated in latest commit.

File: README.md
Comment (chrisbsmith) [reply README.md]: I reworked this as requested and made a few other changes to the configs that align more with the other functions. Also added a few tests to verify the configuration options.

Comment (Darkvater) [reply README.md]: very excited about this. @TwiN have you had time to review the updated PR? If all okay, could we expect a new release soon?

**Comment by TwiN** [comment]
@chrisbsmith Sorry about this, I got distracted 😅 Could you fix the conflict and ping me once you did? I'll get that merged for you ASAP

**Review by TwiN — APPROVED** [review]
Great work!

I'll merge this right away, but if you have some extra time later, an example in https://github.com/TwiN/gatus/tree/master/.examples would be strongly appreciated!

**Comment by chrisbsmith** [comment]
Thanks for the feedback @TwiN! I'm still new to golang and found this as a fun problem to solve and learn.

I didn't even see the `.examples` but will work on one and get another PR when ready.

**Pull request merged**
